src/modules/Users/components/UsersList/UsersList.tsx

```diff
@@ -1,7 +1,305 @@
-import React from 'react'
+import React, { useEffect, useState } from "react";
+import {
+  BASE_IMG_URL,
+  privateAxiosInstance,
+  requestHeader,
+  USERSSURLS,
+} from "../../../../constants/URLS";
+import noPhoto from "../../../../assets/noPhoto.png";
+import "../UsersList/UserList.css";
+import Button from "react-bootstrap/Button";
+import Modal from "react-bootstrap/Modal";
+import { toast } from "react-toastify";
+interface User {
+  id: 1;
+  userName: string;
+  imagePath: null;
+  email: string;
+  country: string;
+  phoneNumber: number;
+  isActivated: boolean;
+  // Add other relevant fields
+}
 
 export default function UsersList() {
+  const [state, setState] = useState("activeId");
+
+  // const userList = async () => {
+  //   const response = await axiosInstance.get(USERSSURLS.getUsersUrl(10,1))
+  //   console.log(response);
+  // };
+
+  const [show, setShow] = useState(false);
+
+  const [userDataById, setUserDataById] = useState<User>({
+    id: 1,
+    userName: "",
+    imagePath: null,
+    email: "string",
+    country: "string",
+    phoneNumber: 0,
+    isActivated: true,
+  })
+
+  const handleClose = () => setShow(false);
+  const getUserById = async (id: number) => {
+    const response = await privateAxiosInstance.get(
+      USERSSURLS.getUserById(id),
+      requestHeader
+    );
+
+    setUserDataById(response.data);
+    setShow(true);
+  };
+
+  // const handleShow = (id: number) => {
+  //   getUserById(id);
+  //   setShow(true);
+  // };
+
+  const [userList, setuserList] = useState([]);
+  const [paginationNumber, setPaginationNumber] = useState<number[]>([0]);
+  const getUserList = async (pageSize: number, pageNumber: number) => {
+    const response = await privateAxiosInstance.get(
+      USERSSURLS.getUsersUrl(pageSize, pageNumber),
+      requestHeader
+    );
+    setPaginationNumber(
+      Array(response.data.totalNumberOfPages)
+        .fill(null)
+        .map((_, i) => i + 1)
+    );
+    setuserList(response.data.data);
+  };
+
+  const blockUser = async (id: number) => {
+    try {
+      await privateAxiosInstance.put(
+        USERSSURLS.toggleStatusUrl(id),
```

Comment:
I don't reccomend adding Url to the object's property as the object name indicates that it has urls.